src/lib/support/DefaultStorageKeyAllocator.h

```diff
@@ -198,11 +198,19 @@ class DefaultStorageKeyAllocator
     }
     static StorageKeyName SubscriptionResumptionMaxCount() { return StorageKeyName::Formatted("g/sum"); }
 
-    static StorageKeyName GlobalSceneCountKey() { return StorageKeyName::Formatted("g/scc"); }
-    static StorageKeyName FabricSceneDataKey(chip::FabricIndex fabric) { return StorageKeyName::Formatted("f/%x/sc", fabric); }
-    static StorageKeyName FabricSceneKey(chip::FabricIndex fabric, uint8_t id)
+    // Scene Count in an Endpoint's SceneTable
+    static StorageKeyName EndpointSceneCountKey(EndpointId endpoint) { return StorageKeyName::Formatted("g/scc/e/%x", endpoint); }
+
+    // General data of all scenes ID belonging to a Fabric on a specific endpoint
+    static StorageKeyName FabricSceneDataKey(FabricIndex fabric, EndpointId endpoint)
+    {
+        return StorageKeyName::Formatted("f/%x/e/%x/sc", fabric, endpoint);
+    }
+
+    // Scene key, Fabric scoped
```

Comment:
What's stored under this key?  How to determine the right value of "idx"?

Reply:
idx goes from 0 to scene max - 1. 
The correct value is held in the scene_map of FabricSceneData, which maps scene_id + group_id to scene idx.
The reason this was made this way is to make it easy to clear all scene from memory in case id's are lost or the number of scene is not coherent between the table count and the endpoint count and etc.

Reply:
Comment added to clarify.